src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -165,21 +170,72 @@ void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) c
     }
 
     // Step 4: issue an error or a warning if the field was not found
-
     // if all sets are missing, we stop the simulation.
     if( areAllSetsMissing )
     {
       // loop again over the map to collect the set names
       string_array missingSetNames;
+
       for( auto const & mapEntry : isTargetSetCreated )
       {
         missingSetNames.emplace_back( mapEntry.first );
       }
-      GEOS_THROW( GEOS_FMT( "\n{}: there is/are no set(s) named `{}` under the {} `{}`.\n",
-                            fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::objectPathString() ),
-                            fmt::join( missingSetNames, ", " ),
-                            FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath() ),
-                  InputError );
+
+      string setNamesError = GEOS_FMT( "\n{}: there is/are no set(s) named `{}` under the {} `{}`.\n",
+                                       fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::objectPathString() ),
+                                       fmt::join( missingSetNames, ", " ),
+                                       FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath() );
+
+      Group const & problemManager = this->getGroupByPath( "/Problem" );
+      string const & objPath = fs.getObjectPath();
+
+      string setNamesAvailable;
+
+      if( objPath == "nodeManager" )
+      {
+        setNamesAvailable.append( "Set names available are: xneg ,yneg ,zneg ,xpos ,ypos, " );
+        stdVector< string > const ff =  problemManager.getGroup( "Mesh" ).getSubGroupsNames();
+
+        if( problemManager.getGroup( "Mesh" ).hasSubGroupOfType< InternalWellboreGenerator >() )
+        {
+          setNamesAvailable.append( ",rneg ,rpos ,tneg ,tpos" );
+        }
+      }
+
+      if( GeometricObjectManager::getInstance().numSubGroups() > 0 )
+      {
+        if( setNamesAvailable.empty())
+          setNamesAvailable.append( "Set names available are: " );
```

Comment:
```suggestion
          setNamesAvailable.append( "Available set names are: " );
```
(and same at line 196)